src/shadowbox/model/access_key.ts

```diff
@@ -62,7 +64,11 @@ export interface AccessKeyRepository {
   // Gets the metrics id for a given Access Key.
   getMetricsId(id: AccessKeyId): AccessKeyMetricsId|undefined;
   // Sets a data transfer limit for all access keys.
-  setAccessKeyDataLimit(limit: DataLimit): Promise<void>;
+  setDefaultDataLimit(limit: DataLimit): Promise<void>;
   // Removes the access key data transfer limit.
-  removeAccessKeyDataLimit(): Promise<void>;
+  removeDefaultDataLimit(): Promise<void>;
+  // Sets access key `id` to use the given custom data limit.
+  setAccessKeyDataLimit(id: AccessKeyId, limit: DataLimit);
+  // Removes the custom data limit from access key `id`.
+  removeAccessKeyDataLimit(id: AccessKeyId);
```

Comment:
Your implementations of `[set|remove]AccessKeyDataLimit` return `Promise<void>`, let's update the interface to reflect that.